### ISSUE TITLE
v1.9.0-rc3 libnetwork sporadic error: failed to create new osl sandbox

### ISSUE THREAD

**Issue by imikushin** [docker/docker#17527]
I'm still troubleshooting it, but here's what I have now.

Trying to get the current RancherOS v0.4.1-dev working with docker-1.9.0-rc3 I'm sporadically getting the following error on one of the system containers, usually after a reboot (but not necessarily):

```
failed to create new osl sandbox: open /var/run/docker/netns/default: permission denied
```

Here's a docker inspect of that container: https://gist.github.com/imikushin/78ccef7309f07b37a00e

I've traced the error to be happening here:
https://github.com/docker/libnetwork/blob/9e400ad567a4d8d76bf69cccd9be75d715b2f2d4/osl/namespace_linux.go#L197

The args here are: `path` == `/var/run/docker/netns/default`, `osCreate` == `false`

Down the call-stack I can see `osCreate` is set to false if we're using `--net=host` which is the case for the failing container. 

`docker version`:
```
[root@rancher-dev ~]# system-docker version
Client:
 Version:      1.9.0-rc3
 API version:  1.21
 Go version:   go1.4.3
 Git commit:   8f66d53
 Built:        Thu Oct 29 15:07:26 UTC 2015
 OS/Arch:      linux/amd64

Server:
 Version:      1.9.0-rc3
 API version:  1.21
 Go version:   go1.4.3
 Git commit:   8f66d53
 Built:        Thu Oct 29 15:07:26 UTC 2015
 OS/Arch:      linux/amd64
```
This is just our zombie reaping patch on top of v1.9.0-rc3: https://github.com/rancher/docker/commits/v1.9.0-rc3-ros1

`docker info`:
```
[root@rancher-dev ~]# system-docker info
Containers: 18
Images: 28
Server Version: 1.9.0-rc3
Storage Driver: overlay
 Backing Filesystem: extfs
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 4.2.3-rancher
Operating System: <unknown> (containerized)
CPUs: 2
Total Memory: 996.3 MiB
Name: rancher-dev
ID: SYVA:H3LL:B2HR:DM4D:UL5K:CICK:AVYY:6EOL:FGWQ:D6WA:DJVK:ITCC
```

`uname -a`:
```
[root@rancher-dev ~]# uname -a
Linux rancher-dev 4.2.3-rancher #1 SMP Wed Oct 14 11:25:04 UTC 2015 x86_64 GNU/Linux
```

Repro: build and boot RancherOS off this branch: https://github.com/imikushin/os/tree/docker-1.9

**Comment by ibuildthecloud** [comment]
@mavenugo This is the issue we are currently work through where 1.9 breaks in RanhcerOS.  I'll try to troubleshoot this more in a couple hours

**Comment by mavenugo** [comment]
@imikushin @ibuildthecloud can you please confirm if this happens on both the daemons that you run ?
If am not wrong RancherOS runs 2 docker daemons in the same machine (1 called as system daemon and the other as user daemon). it seems some permission issue (as the error indicates) accessing it from both the daemons ?

@mrjana do you have any pointers for @imikushin to look at ?

**Comment by imikushin** [comment]
@mavenugo I've only been seeing this in system-docker yet.

**Comment by mrjana** [comment]
I am not sure how all of a sudden accessing /var/run/docker/netns/default suddenly returns EPERM. Can you post your syslog and dmesg? May be apparmor or selinux is active and interfering with the access?

**Comment by ibuildthecloud** [comment]
@tiborvass @mrjana I'm working on this issue now, I'm determined to narrow it down.  I know RancherOS is a weird beast but I don't think it has to do with our two daemon madness because these errors happen before we start the second one.

So @tiborvass knowing my work habits I will probably have a fix at 2am and then demand it get into 1.9 :)

**Comment by ibuildthecloud** [comment]
@mrjana @mavenugo Super easy to reproduce

for i in {1..20}; do (docker run --net=host tianon/true&);done

**Comment by ibuildthecloud** [comment]
@tiborvass c`mon i have a failing tests case, can i get a 1.9.0 milestone

**Comment by ibuildthecloud** [comment]
In all honesty I think this must be fixed before release.  This is just an issue with launching `--net=host` containers in parallel.

**Comment by crosbymichael** [comment]
are we really bind mounting the host's namespace to a default and doing a setns on it?

**Comment by mrjana** [comment]
@crosbymichael We do bind mount host's namespace to a single path but it just happens one for all host containers. And we don't setns on it because we don't do any operations on it

**Comment by crosbymichael** [comment]
why bind mount it if you don't setns on it?

**Comment by mrjana** [comment]
it's just a code path where we don't do any special logic to handle host namespace as special.

**Comment by mrjana** [comment]
@ibuildthecloud Reproduced. Fix on it's way

**Comment by mrjana** [comment]
@ibuildthecloud @imikushin if you have time please try out the code in #17548 which has the fix for this

**Comment by imikushin** [comment]
Might be a separate issue, but @ibuildthecloud 's repro now produces the following errors with docker-1.9.0-dev: 
```
[root@rancher-dev ~]# for i in {1..20}; do (docker run --net=host tianon/true&);done
[root@rancher-dev ~]# invalid argument
invalid argument
invalid argument
Error response from daemon: Cannot start container e50cb92d97a9d94dd7341326d47f461d1f9d2b33bb05d63fb97e0df1f3bbac2e: [8] System error: invalid argument
invalid argument
invalid argument
invalid argument
invalid argument
Error response from daemon: Cannot start container cd8219a3b5269010d4829ff5501e7350a64903dfb238ab4ae3c6fb37f5429b0d: [8] System error: invalid argument
Error response from daemon: Cannot start container 7075155713f2df746e3986a4a13221549c443c29c887d0753b9e15919218a134: [8] System error: invalid argument
Error response from daemon: Cannot start container a5c91ca8ca25593184bffde42a3e7633eae67d81af59e652b5cc83a5f2a5223a: [8] System error: invalid argument
invalid argument
Error response from daemon: Cannot start container 02073cdbd6a612012cc08bc73fbd303dd5d7378fbe604b6fb65d4838af5c2f27: [8] System error: invalid argument
Error response from daemon: Cannot start container c29783880c924a4b9b881df986215a7e47c8066ac3cf624cc93887367e34ef45: [8] System error: invalid argument
invalid argument
Error response from daemon: Cannot start container d3ed8e464f62bd21cc1666a42a3cc3179ff3836ff332050879ed3f2d92949a1b: [8] System error: invalid argument
Error response from daemon: Cannot start container ef37ad40123e6b5ef0ba971ae8ff760b5ae9808104c0b9426029d4ee36f1875c: [8] System error: invalid argument
invalid argument
invalid argument
invalid argument
invalid argument
invalid argument
Error response from daemon: Cannot start container c5715c70ffd8e318671cdb40836f9f5d56d6014b9f62b5556baa5615d882296f: [8] System error: invalid argument
Error response from daemon: Cannot start container 17d6716db1d50e76b90f062edfb0ff22a1114f0ec9f1052b8c3f42c22a6d3d1a: [8] System error: invalid argument
invalid argument
invalid argument
Error response from daemon: Cannot start container a9a26a8ebc971a7ec74ce70ace5af360e1f043e24bd99f3983c14ad8ed6f5701: [8] System error: invalid argument
Error response from daemon: Cannot start container e10d67ea7fac5d478f5d38c6be41d24ce81461c93a6abb663cb6a8c4bf4c2d72: [8] System error: invalid argument
Error response from daemon: Cannot start container e52dbeb20a1d26037dc407d338686704311b98b7f3546f6041168f355293c32e: [8] System error: invalid argument
Error response from daemon: Cannot start container 8921926b0008dd1710bdf224ebe78a51c874a22ceec8e42041bb493eaa4c706c: [8] System error: invalid argument
Error response from daemon: Cannot start container e1f33e1d10e31539d28f7c834d8fd85cd20d6b101a9d6179cafc8d40f28c314c: [8] System error: invalid argument
Error response from daemon: Cannot start container 30d9444c7d9a38411586ea8b1f33eb5f7f0136c5d31fa8503826e36f788e4426: [8] System error: invalid argument
```

Also showed up in one of our system containers after 3 reboots:
```
FATA[0002] 500 Internal Server Error: Cannot start container 96e816caabeab1551810b4f1fd95727ddc073b8be299b54f3025344feec7a55b: [8] System error: invalid argument
```
Here's the docker inspect output: https://gist.github.com/imikushin/858765230a268f0578ff

Just in case, the docker version:
```
[root@rancher-dev ~]# docker version
Client:
 Version:      1.9.0-dev
 API version:  1.22
 Go version:   go1.5.1
 Git commit:   269d4c8
 Built:        Sat Oct 31 10:24:51 UTC 2015
 OS/Arch:      linux/amd64

Server:
 Version:      1.9.0-dev
 API version:  1.22
 Go version:   go1.5.1
 Git commit:   269d4c8
 Built:        Sat Oct 31 10:24:51 UTC 2015
 OS/Arch:      linux/amd64
```
This is our patch on top of docker master: https://github.com/rancher/docker/commits/v1.9.0-dev-ros2
